### PR TITLE
Add Trackable component

### PR DESCRIPTION
**Pull request by patrickkim**

### CODE REVIEW

File: src/atoms/Trackable/Readme.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+Trackable is a helper component that allows you render data attributes onto a Component.
+This is to facilitate third-party analytics scripts and to explicitly expose elements/components that are actively being tracked.
+
+This component:
+- only takes in a `data` object as its property.
+- take key names from the `data` object, lowercases + kebab cases the key name and appends `data-`
```
Comment (drewdrewthis) [inline src/atoms/Trackable/Readme.md]: Typo *takes

**Comment by drewdrewthis** [comment]
👀

File: src/atoms/Trackable/__tests__/trackable.spec.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+import Trackable from 'atoms/Trackable';
+import { dataAttributes, dataAttrName } from 'atoms/Trackable/utils';
+
+describe('<Trackable />', () => {
+
+  describe('dataAttributes()', () => {
+
+    it('takes a string and returns a lower+kebab case prefixed with data-', () => {
+      expect(dataAttrName('SoMe Name')).to.be.equal('data-some-name');
```
Comment (drewdrewthis) [inline src/atoms/Trackable/__tests__/trackable.spec.js]: Is this supposed to be "SoMe"? What happens when you don't use the correct casing?

File: src/atoms/Trackable/index.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { dataAttributes } from './utils';
+
+function wrappedChild(children, data) {
+  const child = React.Children.only(children);
```
Comment (drewdrewthis) [inline src/atoms/Trackable/index.js]: Nice. I was hoping you'd use this. Good call.

File: src/atoms/Trackable/index.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+  if (child.type.displayName === 'Trackable') return child;
```
Comment (drewdrewthis) [inline src/atoms/Trackable/index.js]: Do you have to use `displayName`? I thought you could just reference the type of the function:
`child.type = Trackable`

File: src/atoms/Trackable/index.js (hunk 3 above)
Comment (drewdrewthis) [inline src/atoms/Trackable/index.js]: The only question I have here is, what is the error that gets thrown? Is it possible to provide a clearer error?

File: src/atoms/Trackable/utils.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  return `data-${kebabCase(key.toLowerCase())}`;
+}
+
+export function dataAttributes(data) {
+  const exportAttr = (res, key) => {
+    if (key === 'children') return res;
+
+    const attr = { [dataAttrName(key)]: data[key] };
+
+    return Object.assign({}, res, attr);
```
Comment (drewdrewthis) [inline src/atoms/Trackable/utils.js]: You're already using lodash, why not use their assign method?

File: src/atoms/Trackable/__tests__/trackable.spec.js (hunk 2 above)
Comment (patrickkim) [inline src/atoms/Trackable/__tests__/trackable.spec.js]: just checking that it lowercases all.

Comment (patrickkim) [inline src/atoms/Trackable/__tests__/trackable.spec.js]: BaCk iN tHe DaY ThiS wAs CoOL

File: src/atoms/Trackable/utils.js (hunk 5 above)
Comment (patrickkim) [inline src/atoms/Trackable/utils.js]: Prefer native over libraries.

File: src/atoms/Trackable/index.js (hunk 4 above)
Comment (patrickkim) [inline src/atoms/Trackable/index.js]: 👍

File: src/atoms/Trackable/index.js (hunk 3 above)
Comment (patrickkim) [inline src/atoms/Trackable/index.js]: No error, just doesn't pass through attributes.

**Pull request merged**